test/e2e/clusterctl_upgrade.go

```diff
@@ -387,14 +389,32 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 			input.PreWaitForCluster(managementClusterProxy, testNamespace.Name, workLoadClusterName)
 		}
 
+		// Get the input
+		input := inputGetter()
+
+		// If GroupVersion is not set, default it to v1beta1
+		if input.GroupVersion.Empty() {
```

Comment:
Sorry my bad. Let's simplify the whole thing. Let's just get the CRD and take the group + version from there